Add support for Python 3.11 and PyPy3.9

hugovk opened the pull request:
Changes proposed in this pull request:

* Support Python 3.11:
  * add Trove classifier
  * test on CI
  * build wheel 
* Create wheel for PyPy3.9

[Python 3.11 is currently at release candidate 2](https://discuss.python.org/t/python-3-11-0rc2-is-now-available/18988?u=hugovk), the last one before full release on [2022-10-24](https://peps.python.org/pep-0664/#schedule), and includes this call to action:

> **Community members**
> We strongly encourage maintainers of third-party Python projects to prepare their projects for 3.11 compatibilities during this phase. As always, report any issues to [the Python bug tracker 1](https://github.com/issues).

And:

> There will be no ABI changes from this point forward in the 3.11 series and the goal is that there will be as few code changes as possible.

So it is pretty safe to release a 3.11 release now. We can label it as a dev preview, and put also out a new release when 3.11.0 is out.

But also, if you'd prefer to wait until the full 3.11.0 release, that's fine too, we've not yet had anyone asking for 3.11 wheels yet.

---


Also in this PR:

* Bump GitHub Actions versions
* GHA: Use ubuntu-latest, except pin to latest 22.04 for the benchmarking

hugovk committed:
Bump actions

hugovk committed:
GHA: Use ubuntu-latest (except pin to latest 22.04 for benchmark)

hugovk committed:
Support Python 3.11

hugovk committed:
Extract single matrix factor

hugovk committed:
Create wheel for PyPy3.9

bwoodsend reviewed (approved):
RC2 is close enough to the real thing for me. Bombs away...

JustAnotherArchivist reviewed (approved):
Normal release without a dev preview mark or similar sounds good to me.
Other changes look fine as well, thanks!

JustAnotherArchivist commented:
I'm getting a red X for the CI on b5ce3129 for whatever reason even though all checks passed. Just GitHub Actions being weird?

bwoodsend commented:
A red X which, if you click on it, creates a popup showing that all 36 checks passed. :roll_eyes: Is it just me or is GH's web UI getting buggier and buggier...

hugovk commented:
I think the red X is because I added the "changelog" label after creating the PR:

![image](https://user-images.githubusercontent.com/1324225/190391043-0981e4c8-1e16-4c9a-baae-68d324561e70.png)

So the "Require PR label" action failed, as expected. Then I added it, and it re-ran, and passed, as expected:

![image](https://user-images.githubusercontent.com/1324225/190391095-c591491b-0b67-4537-b63d-4bcccaa6ffad.png)

But it makes no sense to still show the red X if it later passed... 🤷

hugovk commented:
OK, all good to release with these release notes?

![image](https://user-images.githubusercontent.com/1324225/190391455-431939ba-8ad9-4c15-ad0b-47bb9a31a055.png)

https://github.com/ultrajson/ultrajson/releases

bwoodsend commented:
Yay! I did something in this one :smile:
